src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,48 @@
+package main.java.duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a type of task called "Deadline", with a specified end date and time
+ *
+ * @author eugeneleong
+ * @version 1.0
+ */
+
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
+
+    public Deadline(String action, LocalDateTime by) {
+        super(action);
+        this.by = by;
+    }
+
+    /**
+     * Gives the date of the deadline as output
+     * @return date (in dd-MMM-yyyy)
+     */
+    public String getDate() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MMM-yyyy");
```

Comment:
I appreciate your use of our local date-time format rather than the default American format.